src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -103,7 +104,16 @@ func (r *ClusterReconciler) Reconcile(
 	observedNodes := make([]string, 0, len(observedPods.Items))
 	// nolint:gocritic // the copies are necessary for further redpandacluster updates
 	for _, item := range observedPods.Items {
-		observedNodes = append(observedNodes, item.Name)
+		name := fmt.Sprintf("%s%c%s%c%s%c%s",
```

Comment:
let's extract this into something like `svc.DnsName()` also please do `%s.%s.%s.svc.cluster.local`

Reply:
Ok. Changed!